libr/asm/arch/avr/format.c

```diff
@@ -156,10 +158,53 @@ static int formatDisassembledOperand(avrDisassembleContext *context, char *strOp
 			fOptions.addressFieldWidth, context->longAddress);
 		break;
 	case OPERAND_IO_REGISTER:
-		retVal = sprintf(strOperand, "%s%02x",
-			OPERAND_PREFIX_IO_REGISTER,
-			dInstruction.operands[operandNum]);
+	{
+		const char *current_register = NULL;
+		bool is_register_found = false;
+		switch (dInstruction.operands[operandNum])
+		{
+			case 0x3b:
+				current_register = "rampz";
+				is_register_found = true;
+				break;
+			case 0x3c:
+				current_register = "eind";
+				is_register_found = true;
+				break;
+			case 0x3d:
+				current_register = "spl";
+				is_register_found = true;
+				break;
+			case 0x3e:
+				current_register = "sph";
+				is_register_found = true;
+				break;
+			case 0x3f:
+				current_register = "sreg";
+				is_register_found = true;
+				break;
+			case 0x04:
+				current_register = "adcl";
+				is_register_found = true;
+				break;
+			case 0x05:
+				current_register = "adch";
+				is_register_found = true;
+				break;
+			default:
+				retVal = snprintf (strOperand, 5, "0x%x", dInstruction.operands[operandNum]);
+				is_register_found = false;
+				break;
+			break;
+		}
+
+		if (is_register_found) {
```

Comment:
actually, is_register_found is ONLY set to true when current_register is != NULL, so you can remove that variable and half of the lines you wrote there